Resources/Computing/ParallelLibraries/Srun.py

```diff
@@ -0,0 +1,106 @@
+""" srun parallel library
+
+Allow to wrap a job/pilot into a srun call that will execute n similar tasks in parallel.
+To work, srun needs to be used in conjonction with Slurm.
+"""
+import os
+import shutil
+import commands
+import stat
+
+from DIRAC.Resources.Computing.ParallelLibraries.ParallelLibrary import ParallelLibrary
+
+
+class Srun(ParallelLibrary):
+
+  def __init__(self, workingDirectory):
+    super(Srun, self).__init__('srun', workingDirectory)
+
+  def generateWrapper(self, executableFile):
+    """
+    Associate the executable with srun, to execute the same command in parallel on multiple nodes.
+    Wrap it in a new executable file
+
+    :param str executableFile: name of the executable file to run
+    :return str: name of the wrapper that runs the executable via srun
+    """
+    # As the executable will be executed by srun, it needs GRP and OTH execution permission
+    if not os.access(executableFile, 5):
+      os.chmod(executableFile, stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH)
+
+    # Build the script to run the executable in parallel multiple times
+    # srun is the command to execute a task multiple time in parallel
+    # -l option: add the task ID to the output
+    # -k option: do not kill the slurm job if one of the nodes is broken
+    cmd = '#!/bin/bash\n'
+    cmd += 'srun -l -k %s' % executableFile
+    self.log.debug("Command generated by Srun:", "%s" % cmd)
+
+    srunWrapper = os.path.join(self.workingDirectory, 'srunExec.sh')
+    with open(srunWrapper, 'w') as exe:
+      exe.write("%s" % cmd)
+    if not os.access(srunWrapper, 5):
```

Comment:
I guess this is supposed to correspond to:
```suggestion
    if not os.access(srunWrapper, os.R_OK | os.X_OK):
```

Reply:
Is there any reason why `os.<permission>` should be preferred over permission numbers (421)?

Reply:
Just for improving the readability. I wasn't familiar with `os.access` and had to check the docs to realise what the `5` meant.